torch/csrc/autograd/profiler_utils.cpp

```diff
@@ -0,0 +1,201 @@
+#include <torch/csrc/autograd/profiler_utils.h>
+
+namespace torch { namespace autograd { namespace profiler {
+
+const size_t CONV2D_STRIDE = 3;
+const size_t CONV2D_PADDING = 4;
+const size_t CONV2D_DILATION = 5;
+const size_t CONV2D_GROUPS = 6;
+
+static bool validateInput(const at::StringView &op_name, size_t min_size,
+                       const std::vector<c10::IValue>& inputs,
+                       const std::vector<int>& should_be_tensor) {
+  std::stringstream ss;
+  if (inputs.size() < min_size) {
+      ss << "Failed to save extra arguments for flops compuation of op "
+         << op_name.str()
+         << ", min size: " << min_size
+         << ", actual size: " << inputs.size();
+      TORCH_WARN(ss.str());
+      return false;
+  }
+  for (auto index : should_be_tensor) {
+    if (!inputs[index].isTensor()) {
+      ss << "Failed to save extra arguments for flops compuation of op "
+         << op_name.str()
+         << ", input[" << index
+         << "] must be a tensor.";
+      TORCH_WARN(ss.str());
+      return false;
+    }
+  }
+  return true;
+}
+
+std::unordered_map<std::string, c10::IValue> saveExtraArgs(const at::RecordFunction& fn) {
+    // foe specific types of fn, return the saved extra args for computing flops
+    std::unordered_map<std::string, c10::IValue> map;
+    std::vector<c10::IValue> inputs = fn.inputs();
+
+    if (inputs.empty()) {
+      // Input shape is unavailable, return empty map
+      return map;
+    }
+
+    if (fn.name() == at::StringView("aten::conv2d")) {
```

Comment:
I don't think you need StringView casting

Reply:
there's str() method of StringView

Reply:
Fixed by converting at::StringView to std::string and compare with const char*